tests/prov/test_prov_k8s_cortx_upgrade.py

```diff
@@ -372,3 +372,113 @@ def test_42179(self):
         assert_utils.assert_true(resp)
         self.collect_sb = False
         LOGGER.info("--------- Test Completed ---------")
+
+    @pytest.mark.lc
+    @pytest.mark.cortx_upgrade
+    @pytest.mark.tags("TEST-41951")
+    def test_41951(self):
+        """Verify suspend already suspended upgrade."""
+        pull_image_thread_list = []
+        LOGGER.info("Test Started.")
+        LOGGER.info("Step 1: Get installed version.")
+        installed_version = self.upgrade_obj.prov_obj.get_installed_version(
+            self.master_node_list[0])
+        resp = self.upgrade_obj.prov_obj.generate_and_compare_both_version(
+            self.upgrade_image, installed_version)
+        assert_utils.assert_true(resp)
+        # Pull upgrade Images on all worker nodes
+        for each in self.worker_node_list:
+            worker_thread = Thread(target=self.upgrade_obj.prov_obj.pull_cortx_image,
+                                   args=(each,))
+            worker_thread.start()
+            pull_image_thread_list.append(worker_thread)
+        for each in pull_image_thread_list:
+            each.join()
+        que = multiprocessing.Queue()
+        start_upgrade_proc = multiprocessing.Process(target=self.rolling_upgrade, args=(
+            True, que, self.prov_conf["upgrade_start"]))
+        # start the Upgrade
+        LOGGER.info("Step 2: Start the upgrade.")
+        start_upgrade_proc.start()
+        time.sleep(self.prov_conf["sleep_time"])  # Wait to start upgrade_thread thread
+        # Verify the upgrade status.
+        LOGGER.info("Step 3: Verify the upgrade status.")
+        status_resp = self.get_status()
+        LOGGER.info("Status is %s", status_resp)
+        # suspend the upgrade.
+        LOGGER.info("Step 4: Suspend the upgrade.")
+        suspend_resp = self.upgrade_obj.upgrade_software(self.master_node_list[0],
```

Comment:
Preferably the name should indicate the operation as `suspend_upgrade` or something similar.